packages/cli/test/unit/services/redis.service.test.ts

```diff
@@ -4,6 +4,21 @@ import config from '@/config';
 import { RedisService } from '@/services/redis.service';
 import { mockInstance } from '../../shared/mocking';
 
+jest.mock('ioredis', () => {
+	const Redis = require('ioredis-mock');
+	if (typeof Redis === 'object') {
+		// the first mock is an ioredis shim because ioredis-mock depends on it
+		// https://github.com/stipsan/ioredis-mock/blob/master/src/index.js#L101-L111
+		return {
+			Command: { _transformer: { argument: {}, reply: {} } },
+		};
+	}
+	// second mock for our code
+	return function (...args: any) {
+		return new Redis(args);
+	};
+});
+
```

Comment:
same here I think

Reply:
This one didn't work so I restored the legacy mock. I find this whole hierarchy of Redis classes convoluted `RedisService → RedisServicePubSubSubscriber RedisServiceBaseSender → RedisServiceBase` but since it's a single test I'd rather not spend longer here right now.

Reply:
I can send a PR cleaning up these old mocks 👍🏽

Reply:
Thanks!! Can be a follow-up also, I'd like to bring this in to unblock the rest of the story.